Client/Frontend/InternalSchemeHandler/SessionRestoreHandler.swift

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import GCDWebServers
+import Shared
+
+func generateResponseThatRedirects(toUrl url: URL) -> (URLResponse, Data) {
+    let startTags = "<!DOCTYPE html><html><head><script>"
+    let endTags = "</script></head></html>"
```

Comment:
I believe for this to be considered "valid", you at least need to have an empty `<body></body>` in there too.